cargo-marker/src/lints.rs

```diff
@@ -6,22 +6,51 @@ use std::{
 
 use crate::ExitStatus;
 
+pub struct LintCrateSpec<'a> {
```

Comment:
Using an extra struct is more readable than the alternative of using a tuple. It should also make it easier to add crates via git etc. Could you add doc comments, explaining why and when package name can be `None`? Also, mentioning that we require the package name to be UTF-8 could be useful. TOML, forces this, but still worth a mention. That will make later audits a bit easier. (If that ever happens :))

Reply:
> Also, mentioning that we require the package name to be UTF-8 could be useful

`str` implies that, so I feel that would be more explicit than it should be. 

> Could you add doc comments, explaining why and when package name can be `None`?

Sure, I'll add some doc comments

Reply:
> str implies that, so I feel that would be more explicit than it should be.

Well, if someone else reads it, it could be good to know that this was an intentional choice. Using stings for paths and file names, is quite common even if it's not always correct. There is #67 because I didn't think about that during the first implementation. That's also why I noticed it.